### PR TITLE
Ifpack2: updated block relaxation options in user's guide (Fix #971)

### LINKED ISSUE #971: Ifpack2: update user guide with new block relaxation options
@trilinos/ifpack2 The Ifpack2 user's guide should be updated to reflect changes in options to block relaxation.

### PR DESCRIPTION
**Pull request by brian-kelley**
Also fixed the actual default BlockRelaxation container type
(set in getValidParameters()) from "TriDi" to "Dense" to match the documentation,
comments and value set in the constructor.

@trilinos/ifpack2 

## Description

In options.tex for the Ifpack2 user's guide, added brief documentation for the BlockRelaxation parameters that were missing. Did not add the "schwarz: ..." options, which are valid (added to list in BlockRelaxation::getValidParameters()) but whose values are never used, so there is no reason for users to care about them.

## Motivation and Context

The section of the user's guide about BlockRelaxation was out of date and it was necessary to look at source code to find the full set of valid parameters and their meanings.

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

**Comment by brian-kelley** [comment]
@aprokop @csiefer2 
1. I just realized that the default container should be TriDi and not Dense, since the place where "relaxation: container" is actually set is getValidParameters(), and that was made to be TriDi much more recently than any mention of Dense (this year vs 2016). Even though containerType_ was set to Dense in the constructor, it's overwritten with TriDi when setParameters() is called. So now I've fixed the out-of-date comment and the value set in the constructor. This change doesn't actually make the code behave differently.

2. It would be an easy code change, but I assume there are users out there who have "TriDi" in their parameter lists.

**Comment by mhoemmen** [comment]
@brian-kelley wrote:
> I assume there are users out there who have "TriDi" in their parameter lists.

You could accept both :)

**Comment by brian-kelley** [comment]
@aprokop @mhoemmen I added "Tridiagonal" as an alternative ("TriDi" still works), and updated the user's guide accordingly. Should I do the same for "BlockTriDi?" I also added a bit of information to the user's guide about how each container does the solve.

**Comment by mhoemmen** [comment]
@brian-kelley wrote:
> Should I do the same for "BlockTriDi?"

Yes please, if you have time: "Block Tridiagonal".  Thanks for working on this! :D

**Comment by mhoemmen** [comment]
@trilinos/framework This looks like an Intel license server failure:
```
CMake Error at /projects/sems/install/rhel6-x86_64/atdm/binary-install/cmake-3.11.1-Linux-x86_64/share/cmake-3.11/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "/projects/sems/install/rhel6-x86_64/sems/compiler/intel/17.0.1/mpich/3.2/bin/mpicc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_intel_17.0.1/pull_request_test/CMakeFiles/CMakeTmp
    
    Run Build Command:"/projects/sems/install/rhel6-x86_64/atdm/utility/ninja_fortran/1.7.2/bin/ninja" "cmTC_57c81"
    [1/2] Building C object CMakeFiles/cmTC_57c81.dir/testCCompiler.c.o
    FAILED: CMakeFiles/cmTC_57c81.dir/testCCompiler.c.o 
    /projects/sems/install/rhel6-x86_64/sems/compiler/intel/17.0.1/mpich/3.2/bin/mpicc    -o CMakeFiles/cmTC_57c81.dir/testCCompiler.c.o   -c testCCompiler.c
    
    Error: A license for Comp-CL is not available now (-15,570,115).
    
    A connection to the license server could not be made.  You should
    make sure that your license daemon process is running: both an
    lmgrd process and an INTEL process should be running
    if your license limits you to a specified number of licenses in use
    at a time.  Also, check to see if the wrong port@host or the wrong
    license file is being used, or if the port or hostname in the license
    file has changed.
    
    License file(s) used were (in this order):
        1.  Trusted Storage
    **  2.  /projects/sems/install/rhel6-x86_64/sems/compiler/intel/17.0.1/base/Licenses/intel-Linux-SON.lic
    **  3.  /projects/sems/install/rhel6-x86_64/sems/compiler/intel/17.0.1/base/compilers_and_libraries_2017.1.132/linux/bin/intel64/../../Licenses
    **  4.  /ascldap/users/trilinos/Licenses
    **  5.  /opt/intel/licenses
    **  6.  /Users/Shared/Library/Application Support/Intel/Licenses
    **  7.  /projects/sems/install/rhel6-x86_64/sems/compiler/intel/17.0.1/base/compilers_and_libraries_2017.1.132/linux/bin/intel64/*.lic
    
    Please refer http://software.intel.com/sites/support/ for more information..
    
    icc: error #10052: could not checkout FLEXlm license
    ninja: build stopped: subcommand failed.
```

**Comment by brian-kelley** [comment]
@mhoemmen It looks like there's also build failures in MueLu and Zoltan2, and test failures in SEACAS.

File: packages/ifpack2/doc/UsersGuide/options.tex [hunk 1]
```diff
@@ -179,7 +179,7 @@ \section{Block relaxation}\label{s:block_relaxation}
      ``Amesos2 solver name'' (see the Amesos2 documentation for all available solvers).
 
      If experimental kokkos-kernels features are enabled (true by default), the ``BlockTriDi''
```
Comment (srajama1) [inline packages/ifpack2/doc/UsersGuide/options.tex]: I am not sure why is this still experimental ? This is used by an application.

**Comment by brian-kelley** [comment]
@srajama1 I don't know either. The exact CMake option that enables this is Ifpack2_ENABLE_Experimental_KokkosKernels_Features. Should I just remove the mention of "experimental" in the docs? I guess if a user knows what they're doing and want it off for some reason, they can dig into the code and CMakeLists to figure out how to disable it.

**Comment by srajama1** [comment]
@brian-kelley It is ok to leave it experimental for now. Please check with Ifpack2 leads and create an issue to remove this experimental status.

**Pull request merged**